src/cbc_sdk/endpoint_standard/base.py

```diff
@@ -258,9 +258,77 @@ def __init__(self, cb, model_unique_id=None, initial_data=None, force_init=False
 
+    def get_details(self, timeout=0, async_mode=False):
+        """Requests detailed results.
+
+        Args:
+            timeout (int): Event details request timeout in milliseconds.
+            async_mode (bool): True to request details in an asynchronous manner.
+
+        Note:
+            - When using asynchronous mode, this method returns a python future.
+              You can call result() on the future object to wait for completion and get the results.
+        """
+        self._details_timeout = timeout
+        if not self.event_id:
+            raise ApiError("Trying to get event details on an invalid event_id")
+        if async_mode:
+            return self._cb._async_submit(lambda arg, kwarg: self._get_detailed_results())
+        else:
+            return self._get_detailed_results()
+
+    def _get_detailed_results(self):
+        """Actual search details implementation"""
+        args = {"event_ids": [self.event_id]}
+        url = "/api/investigate/v2/orgs/{}/enriched_events/detail_jobs".format(self._cb.credentials.org_key)
+        query_start = self._cb.post_object(url, body=args)
+        job_id = query_start.json().get("job_id")
+        timed_out = False
+        submit_time = time.time() * 1000
+
+        while True:
+            status_url = "/api/investigate/v2/orgs/{}/enriched_events/detail_jobs/{}".format(
+                self._cb.credentials.org_key,
+                job_id,
+            )
+            result = self._cb.get_object(status_url)
+            searchers_contacted = result.get("contacted", 0)
+            searchers_completed = result.get("completed", 0)
+            log.debug("contacted = {}, completed = {}".format(searchers_contacted, searchers_completed))
+            if searchers_contacted == 0:
+                time.sleep(.5)
+                continue
+            if searchers_completed < searchers_contacted:
+                if self._details_timeout != 0 and (time.time() * 1000) - submit_time > self._details_timeout:
+                    timed_out = True
+                    break
+            else:
+                break
+
+            time.sleep(.5)
+
+        if timed_out:
+            raise TimeoutError(message="user-specified timeout exceeded while waiting for results")
+
+        log.debug("Pulling detailed results, timed_out={}".format(timed_out))
+
+        still_fetching = True
+        result_url = "/api/investigate/v2/orgs/{}/enriched_events/detail_jobs/{}/results".format(
+            self._cb.credentials.org_key,
+            job_id
+        )
+        query_parameters = {}
+        while still_fetching:
+            result = self._cb.get_object(result_url, query_parameters=query_parameters)
+            total_results = result.get('num_available', 0)
+            if total_results != 0:
+                results = result.get('results', [])
+                return EnrichedEvent(self._cb, initial_data=results[0])
```

Comment:
Why return a new model instead of modifying the existing model and returning self?

Reply:
```
self._info = results[0]
return self
```

Reply:
That was actually what bothered me about this...it felt as if I should not return a modified self, but rather the details of a search, then again when I return a new model instane, you could call get_details() on it and it makes no sense. I don't really know what would be best in that case :(

Reply:
So in essence you are not modifying the object but rather enhancing it with additional details. The details search only fetches additional fields and any that might have changed since the original time the event was fetched. You have a point about having already fetched the details do you want to block a second fetch. I'm not opposed but then again that wouldn't be common and it's not the end of the world so it might be better left alone.

Reply:
OK, modifying the object rather than instantiating a new one now.